### PR TITLE
Smockit testing

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
To enable easier testing, smockit can be used to mock contracts.

# changes
- adds smockit
- uses smockit in Insurance tests
- implement insurance tests
- altered comment in Insurance
- small bug fix in `InsurancePoolToken.sol`

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -14,6 +14,11 @@ module.exports = {
                 enabled: true,
                 runs: 1000,
             },
+            outputSelection: {
+                "*": {
+                    "*": ["storageLayout"],
+                },
+            },
```
Comment (CalabashSquash) [inline hardhat.config.js]: what does this do?

File: test/unit/Insurance.js [hunk 2]
```diff
@@ -1,101 +1,390 @@
...
+        const tracerContractFactory = await ethers.getContractFactory(
+            "TracerPerpetualSwaps",
+            {
+                libraries: {
+                    Balances: libBalances.address,
+                    Perpetuals: libPerpetuals.address,
+                    Prices: libPrices.address,
+                },
+            }
+        )
```
Comment (CalabashSquash) [inline test/unit/Insurance.js]: These should be in `deploy/FullDeploy.js` now

**Review by CalabashSquash — CHANGES_REQUESTED** [review]
:bolivia:

File: hardhat.config.js (hunk 1 above)
Comment (raymogg) [reply hardhat.config.js]: Smockit needs access to the internal storage layout to help it mock calls, and this enables that. See: https://github.com/ethereum-optimism/optimism/tree/develop/packages/smock

File: test/unit/Insurance.js (hunk 2 above)
Comment (raymogg) [reply test/unit/Insurance.js]: @BenjaminPatch including the mock contract and what not?

File: hardhat.config.js (hunk 1 above)
Comment (raymogg) [reply hardhat.config.js]: Also I was wrong on the above, its only needed for `smoddit` which we don't use yet. Have removed this config for now

**Review by CalabashSquash — APPROVED** [review]

**Pull request merged**